src/freertos_drivers/ti/TivaEEPROMFile.hxx

```diff
@@ -0,0 +1,103 @@
+/** \copyright
+ * Copyright (c) 2022, Balazs Racz
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file TivaEEPROMFile.hxx
+ *
+ * Implementation for persistent storage that uses Tiva EEPROM.
+ *
+ * @author Balazs Racz
+ * @date 9 Dec 2022
+ */
+
+#ifndef _FREERTOS_DRIVER_TI_TIVAEEPROMFILE_HXX_
+#define _FREERTOS_DRIVER_TI_TIVAEEPROMFILE_HXX_
+
+#include "driverlib/rom.h"
+#include "driverlib/rom_map.h"
+
+#include "driverlib/eeprom.h"
+#include "driverlib/sysctl.h"
+
+#include "freertos_drivers/common/DeviceFile.hxx"
+
+class TivaEEPROMFile : public DeviceFile
+{
+public:
+    TivaEEPROMFile(const char *name, unsigned byte_offset, unsigned byte_size)
+        : DeviceFile(name)
+        , byteOffset_(byte_offset)
+        , byteSize_(byte_size)
+    {
+        MAP_SysCtlPeripheralEnable(SYSCTL_PERIPH_EEPROM0);
+        MAP_EEPROMInit();
```

Comment:
There is an errata on ROM_EEPROMInit(). Use EEPROMInit() without the MAP_ prefix to ensure that the ROM version never gets linked in. When making this change, please add an inline comment referencing the Errata.

Reply:
done